packages/kbn-securitysolution-autocomplete/src/field_value_match/index.tsx

```diff
@@ -120,6 +125,13 @@ export const AutocompleteFieldMatchComponent: React.FC<AutocompleteFieldMatchPro
     [setError, onError]
   );
 
+  const handleWarning = useCallback(
+    (warn: Warning | undefined): void => {
+      onWarning(warn !== undefined);
```

Comment:
So if one were to pass `onWarning` to this component, but not `warning`, the parent `setWarningsExist` is always going to be `false` and nothing will happen, right? Is that the intended behavior?

The control flow here is a little odd; if the component is accepting `onWarning` I would expect the rendering of the warning to be the parent's responsibility, but it seems like it's not (or is `helpText={warning}` something else?).

I guess this all works for now, so nothing's _needed_ to be done, but I wonder if this architecture might be causing some extra renders 🤔